elm-src/index.elm

```diff
@@ -0,0 +1,489 @@
+                    , td [] []
+                    , td [ class "example" ] [ null_is_not_given nix option.example ]
+                    ]
+                , tr []
+                    [ th [] [ text "Declared In:" ]
+                    , td [] []
+                    , td [ class "declared-in" ] (List.map declaration_link option.declarations)
+                    ]
+                ]
+            ]
+        ]
+
+
+optionToTd : Maybe String -> ( String, Option ) -> List (Html Msg)
+optionToTd selected ( name, option ) =
+    let
+        isSelected =
+            (case selected of
+                Nothing ->
+                    False
+
+                Just selopt ->
+                    selopt == name
+            )
+    in
+        (List.concat
+            [ [ (tr []
+                    [ td
+                        [ onClick
+                            (if isSelected then
+                                (DeselectOption)
+                             else
+                                (SelectOption name)
+                            )
+                        ]
+                        [ tt [] [ text name ] ]
+                    ]
+                )
+              ]
+            , (if not isSelected then
+                []
+               else
+                [ tr []
+                    [ td [ class "details" ] [ describe_option option ]
+                    ]
+                ]
+              )
+            ]
+        )
+
+
+opt_tuple_matches_name : String -> ( String, Option ) -> Bool
+opt_tuple_matches_name term ( name, option ) =
+    term == name
+
+
+changePageIf : Bool -> Int -> String -> Html Msg
+changePageIf cond page txt =
+    button
+        (if cond then
+            [ onClick (ChangePage page) ]
+         else
+            [ disabled True ]
+        )
+        [ text txt ]
+
+
+view : Model -> Html Msg
+view model =
+    if List.length model.options > 0 then
+        viewOptions model
+    else
+        div [] [ h1 [] [ text (toString model.status) ] ]
+
+
+debounce =
+    Debounce.debounceCmd debounce_cfg
+
+
+debounce_cfg : Debounce.Config Model Msg
+debounce_cfg =
+    Debounce.config
+        .debounce_state
+        -- getState   : Model -> Debounce.State
+        (\model s -> { model | debounce_state = s })
+        -- setState   : Debounce.State -> Model -> Debounce.State
+        Bounce
+        -- msgWrapper : Msg a -> Msg
+        200
+
+
+
+-- timeout ms : Float
+
+
+viewOptions : Model -> Html Msg
+viewOptions model =
+    div []
+        [ p []
+            [ input [ id "search", class "search-query span3", autofocus True, value model.query, onInput ChangeQuery ] []
+            ]
+        , hr [] []
+        , p [ id "how-many" ]
+            [ em []
+                [ text
+                    (if (List.length model.matchingOptions) == 0 then
+                        "Showing no results"
+                     else
+                        (String.concat
```

Comment:
Specific reason you seem to (generally) prefer `List.concat` and `String.concat` over the `++` concatenation operator? `++` feels little more idiomatic, to me.

Reply:
Ignorance :)